Cesium3DTilesSelection/src/TilesetJsonLoader.cpp

```diff
@@ -0,0 +1,855 @@
+                  pCompletedRequest) mutable {
+            auto pResponse = pCompletedRequest->response();
+            const std::string& tileUrl = pCompletedRequest->url();
+            if (!pResponse) {
+              SPDLOG_LOGGER_ERROR(
+                  pLogger,
+                  "Did not receive a valid response for tile content {}",
+                  tileUrl);
+              return TileLoadResult{
+                  TileUnknownContent{},
+                  TileLoadResultState::Failed,
+                  std::move(pCompletedRequest),
+                  {}};
+            }
+
+            uint16_t statusCode = pResponse->statusCode();
+            if (statusCode != 0 && (statusCode < 200 || statusCode >= 300)) {
+              SPDLOG_LOGGER_ERROR(
+                  pLogger,
+                  "Received status code {} for tile content {}",
+                  statusCode,
+                  tileUrl);
+              return TileLoadResult{
+                  TileUnknownContent{},
+                  TileLoadResultState::Failed,
+                  std::move(pCompletedRequest),
+                  {}};
+            }
+
+            // find gltf converter
+            const auto& responseData = pResponse->data();
+            auto converter = GltfConverters::getConverterByMagic(responseData);
+            if (!converter) {
+              converter = GltfConverters::getConverterByFileExtension(tileUrl);
+            }
+
+            if (converter) {
+              // Convert to gltf
+              CesiumGltfReader::GltfReaderOptions gltfOptions;
+              gltfOptions.ktx2TranscodeTargets =
+                  contentOptions.ktx2TranscodeTargets;
+              GltfConverterResult result = converter(responseData, gltfOptions);
+
+              // Report any errors if there are any
+              logErrorsAndWarnings(pLogger, tileUrl, result.errors);
+              if (result.errors) {
+                return TileLoadResult{
+                    TileRenderContent{std::nullopt},
+                    TileLoadResultState::Failed,
+                    std::move(pCompletedRequest),
+                    {}};
+              }
+
+              return TileLoadResult{
+                  TileRenderContent{std::move(result.model)},
+                  TileLoadResultState::Success,
+                  std::move(pCompletedRequest),
+                  {}};
+            } else {
+              // not a renderable content, then it must be external tileset
+              return parseExternalTilesetInWorkerThread(
+                  tileTransform,
+                  pLogger,
+                  std::move(pCompletedRequest),
+                  std::move(externalContentInitializer));
+            }
+          });
+}
+
+const std::string& TilesetJsonLoader::getBaseUrl() const noexcept {
+  return _baseUrl;
+}
+
+void TilesetJsonLoader::addChildLoader(
+    std::unique_ptr<TilesetContentLoader> pLoader) {
+  _children.emplace_back(std::move(pLoader));
```

Comment:
Most of our existing code uses `this->` to acccess fields. That's not necessary, of course, but I think it adds clarity. I can be convinced to stop doing that (it's admittedly uncommon in C++), but if we go that way we should change all our code (not in this PR!). Because mixing the two patterns is kind of the worst of all worlds.

Reply:
I updated to use this keyword to access class field in the unit test PR. I leave this comment here to mark this as temporarily resolved until the PR is merged